src/App.vala

```diff
@@ -86,6 +86,37 @@ public class Dock.App : Object {
         }
     }
 
+    public bool launch_new_instance () {
+        var single_main_window = app_info.get_string ("SingleMainWindow");
+        if (single_main_window == "true") {
+            return false;
+        }
+
+        var context = Gdk.Display.get_default ().get_app_launch_context ();
+        context.set_timestamp (Gdk.CURRENT_TIME);
```

Comment:
I believe it's better to pass the context to the function from the event controller handler. We can get the display from the widget itself and set a valid timestamp with `Gtk.EventController.get_current_event_time()`.